src/syrupy/extensions/single_file.py

```diff
@@ -65,17 +80,37 @@ def _read_snapshot_data_from_location(
         self, *, snapshot_location: str, snapshot_name: str
     ) -> Optional["SerializableData"]:
         try:
-            with open(snapshot_location, "rb") as f:
+            with open(
```

Comment:
You guys'll get a kick out of this. We upgraded from 1.7.3 to 3.0.X and some of our tests with json snapshots started failing. The snapshots updated fine, but were returning the wrong thing when reading the snapshots.

Turns out we were decorating the test with @mock.patch('builtins.open') rather than constricting it to the code it actually needed the mock for. This used to work with Json snapshots because it was using read_text. Oops!